connect/src/main/java/com/stripe/android/connect/webview/StripeConnectWebViewClient.kt

```diff
@@ -39,6 +43,29 @@ internal class StripeConnectWebViewClient(
     }
 ) : WebViewClient() {
 
+    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+        view ?: return false // the view shouldn't be null, but if it is then don't handle the request
+        val url = request?.url ?: return false // if the request isn't for a url, then don't handle it
+        return if (url.host in ALLOWLISTED_HOSTS) {
+            logger.warning("(StripeConnectWebViewClient) Received pop-up for allow-listed host: $url")
+            false // Allow the request to propagate so we open URL in WebView, but this is not an expected operation
+        } else if (url.scheme == "https" || url.scheme == "http") {
+            // open the URL in an external browser for safety and to preserve back navigation
+            logger.debug("(StripeConnectWebViewClient) Opening URL in external browser: $url")
+            launchChromeCustomTab(view.context, url)
+            true // block the request since we're opening it in a chrome tab
+        } else {
+            // don't launch if it's an unsupported scheme (ie. not http/https)
+            logger.debug("(StripeConnectWebViewClient) Received unsupported pop-up request: $url")
+            true // block the request as it's unsupported
```

Comment:
Is this getting handled in [MXMOBILE-2914](https://jira.corp.stripe.com/browse/MXMOBILE-2914)? If so, should we add a TODO for it?

We have legitimate use cases for mailto urls, so we will need to handle this. A TODO would help clarify we're planning to do this in a followup PR 😄

Reply:
Yeah, good call